### ISSUE TITLE
Failed to load the pretrained model from the dbnet++ official release

### ISSUE THREAD

**Issue by EYcab** [open-mmlab/mmocr#1021]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
The multi-gpus training on dbnet++ is good to run on my dataset but failed to load the pretrained model from the official release:
dbnetpp_r50dcnv2_fpnc_1200e_icdar2015-20220502-d7a76fff.pth
from https://github.com/open-mmlab/mmocr/blob/main/configs/textdet/dbnetpp/README.md

**Reproduction**
2022-05-18 20:52:34,167 - mmocr - INFO - Set random seed to 1546938340, deterministic: False
2022-05-18 20:52:34,410 - mmocr - INFO - initialize ResNet with init_cfg {'type': 'Pretrained', 'checkpoint': 'torchvision://resnet50'}
2022-05-18 20:52:34,410 - mmcv - INFO - load model from: torchvision://resnet50
2022-05-18 20:52:34,410 - mmcv - INFO - load checkpoint from torchvision path: torchvision://resnet50
2022-05-18 20:52:34,566 - mmcv - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: fc.weight, fc.bias

missing keys in source state_dict: layer2.0.conv2.conv_offset.weight, layer2.0.conv2.conv_offset.bias, layer2.1.conv2.conv_offset.weight, layer2.1.conv2.conv_offset.bias, layer2.2.conv2.conv_offset.weight, layer2.2.conv2.conv_offset.bias, layer2.3.conv2.conv_offset.weight, layer2.3.conv2.conv_offset.bias, layer3.0.conv2.conv_offset.weight, layer3.0.conv2.conv_offset.bias, layer3.1.conv2.conv_offset.weight, layer3.1.conv2.conv_offset.bias, layer3.2.conv2.conv_offset.weight, layer3.2.conv2.conv_offset.bias, layer3.3.conv2.conv_offset.weight, layer3.3.conv2.conv_offset.bias, layer3.4.conv2.conv_offset.weight, layer3.4.conv2.conv_offset.bias, layer3.5.conv2.conv_offset.weight, layer3.5.conv2.conv_offset.bias, layer4.0.conv2.conv_offset.weight, layer4.0.conv2.conv_offset.bias, layer4.1.conv2.conv_offset.weight, layer4.1.conv2.conv_offset.bias, layer4.2.conv2.conv_offset.weight, layer4.2.conv2.conv_offset.bias

**Comment by xinke-wang** [comment]
Hi, according to your log ` initialize ResNet with init_cfg {'type': 'Pretrained', 'checkpoint': 'torchvision://resnet50'}`, this warning does not related to the dbnet++ pretrained model. Can you share your config here?

**Comment by EYcab** [comment]
log_config = dict(interval=5, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'demo/exp/dbnetpp/dbnetpp_r50dcnv2_fpnc_100k_iter_synthtext-20220502-db297554.pth'
resume_from = None
workflow = [('train', 1)]
opencv_num_threads = 0
mp_start_method = 'fork'
optimizer = dict(type='Adam', lr=0.001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(policy='step', gamma=0.3, step=[5, 10, 40, 75])
total_epochs = 100
model = dict(
    type='DBNet',
    backbone=dict(
        type='mmdet.ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=-1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        dcn=dict(type='DCNv2', deform_groups=1, fallback_on_stride=False),
        init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50'),
        stage_with_dcn=(False, True, True, True)),
    neck=dict(
        type='FPNC',
        in_channels=[256, 512, 1024, 2048],
        lateral_channels=256,
        asf_cfg=dict(attention_type='ScaleChannelSpatial')),
    bbox_head=dict(
        type='DBHead',
        in_channels=256,
        loss=dict(type='DBLoss', alpha=5.0, beta=10.0, bbce_loss=True),
        postprocessor=dict(
            type='DBPostprocessor', text_repr_type='quad',
            epsilon_ratio=0.002)),
    train_cfg=None,
    test_cfg=None)

#Above is a key part of the config.

**Comment by gaotongxiao** [comment]
Your config looks fine to me and it shouldn't cause any error. Did you modify anything else? What was the command you used to launch testing?

**Comment by EYcab** [comment]
I barely use the command in the site： https://mmocr.readthedocs.io/en/latest/training.html
./tools/dist_train.sh ${CONFIG_FILE} ${WORK_DIR} ${GPU_NUM} [PY_ARGS]

It could train as usual but failed to load the model as described above.

**Comment by EYcab** [comment]
unexpected key in source state_dict: fc.weight, fc.bias

missing keys in source state_dict: layer2.0.conv2.conv_offset.weight, layer2.0.conv2.conv_offset.bias, layer2.1.conv2.conv_offset.weight, layer2.1.conv2.conv_offset.bias, layer2.2.conv2.conv_offset.weight, layer2.2.conv2.conv_offset.bias, layer2.3.conv2.conv_offset.weight, layer2.3.conv2.conv_offset.bias, layer3.0.conv2.conv_offset.weight, layer3.0.conv2.conv_offset.bias, layer3.1.conv2.conv_offset.weight, layer3.1.conv2.conv_offset.bias, layer3.2.conv2.conv_offset.weight, layer3.2.conv2.conv_offset.bias, layer3.3.conv2.conv_offset.weight, layer3.3.conv2.conv_offset.bias, layer3.4.conv2.conv_offset.weight, layer3.4.conv2.conv_offset.bias, layer3.5.conv2.conv_offset.weight, layer3.5.conv2.conv_offset.bias, layer4.0.conv2.conv_offset.weight, layer4.0.conv2.conv_offset.bias, layer4.1.conv2.conv_offset.weight, layer4.1.conv2.conv_offset.bias, layer4.2.conv2.conv_offset.weight, layer4.2.conv2.conv_offset.bias

2022-05-18 20:52:34,591 - mmocr - INFO - initialize FPNC with init_cfg [{'type': 'Kaiming', 'layer': 'Conv'}, {'type': 'Constant', 'layer': 'BatchNorm', 'val': 1.0, 'bias': 0.0001}]
2022-05-18 20:52:34,610 - mmocr - INFO - initialize ScaleChannelSpatialAttention with init_cfg [{'type': 'Kaiming', 'layer': 'Conv', 'bias': 0}]
2022-05-18 20:52:34,612 - mmocr - INFO - initialize DBHead with init_cfg [{'type': 'Kaiming', 'layer': 'Conv'}, {'type': 'Constant', 'layer': 'BatchNorm', 'val': 1.0, 'bias': 0.0001}]
2022-05-18 20:52:42,455 - root - INFO - ModulatedDeformConvPack backbone.layer2.0.conv2 is upgraded to version 2.
2022-05-18 20:52:42,456 - root - INFO - ModulatedDeformConvPack backbone.layer2.1.conv2 is upgraded to version 2.
2022-05-18 20:52:42,457 - root - INFO - ModulatedDeformConvPack backbone.layer2.2.conv2 is upgraded to version 2.
2022-05-18 20:52:42,458 - root - INFO - ModulatedDeformConvPack backbone.layer2.3.conv2 is upgraded to version 2.
2022-05-18 20:52:42,459 - root - INFO - ModulatedDeformConvPack backbone.layer3.0.conv2 is upgraded to version 2.
2022-05-18 20:52:42,461 - root - INFO - ModulatedDeformConvPack backbone.layer3.1.conv2 is upgraded to version 2.
2022-05-18 20:52:42,462 - root - INFO - ModulatedDeformConvPack backbone.layer3.2.conv2 is upgraded to version 2.
2022-05-18 20:52:42,463 - root - INFO - ModulatedDeformConvPack backbone.layer3.3.conv2 is upgraded to version 2.
2022-05-18 20:52:42,465 - root - INFO - ModulatedDeformConvPack backbone.layer3.4.conv2 is upgraded to version 2.
2022-05-18 20:52:42,466 - root - INFO - ModulatedDeformConvPack backbone.layer3.5.conv2 is upgraded to version 2.
2022-05-18 20:52:42,468 - root - INFO - ModulatedDeformConvPack backbone.layer4.0.conv2 is upgraded to version 2.
2022-05-18 20:52:42,471 - root - INFO - ModulatedDeformConvPack backbone.layer4.1.conv2 is upgraded to version 2.
2022-05-18 20:52:42,474 - root - INFO - ModulatedDeformConvPack backbone.layer4.2.conv2 is upgraded to version 2.
2022-05-18 20:52:42,481 - mmocr - INFO - Start running, host: root@semantictranslationgpu03-prod-idcb, work_dir: /data/mmocr_v4/demo/exp/funsd_textocr_wzocr_0519
2022-05-18 20:52:42,481 - mmocr - INFO - Hooks will be executed in the following order:
before_run:
(VERY_HIGH   ) StepLrUpdaterHook
(NORMAL      ) CheckpointHook
(NORMAL      ) DistEvalHook
(VERY_LOW    ) TextLoggerHook
 --------------------
before_train_epoch:
(VERY_HIGH   ) StepLrUpdaterHook
(NORMAL      ) DistSamplerSeedHook
(NORMAL      ) DistEvalHook
(LOW         ) IterTimerHook
(VERY_LOW    ) TextLoggerHook
 --------------------
before_train_iter:
(VERY_HIGH   ) StepLrUpdaterHook
(NORMAL      ) DistEvalHook
(LOW         ) IterTimerHook
 --------------------

**Comment by EYcab** [comment]
I guess there is some clues from the line in the report:
2022-05-18 20:52:42,455 - root - INFO - ModulatedDeformConvPack backbone.layer2.0.conv2 is upgraded to version 2.

Have you happened to encounter this before?

**Comment by gaotongxiao** [comment]
The key-missing error comes from the initialization stage where MMOCR tries to load ResNet50 released by PyTorch, whilst ResNet50-dcn used by DBNet++ has additional dcn layers. However, all the weights, including these "missing" ones, will be refreshed by pretrained weights in the `load-from` stage that happens later, as the log indicates. So, please don't worry, everything works well.
